src/frontends/tensorflow_common/src/op/adjust_saturation.cpp

```diff
@@ -0,0 +1,185 @@
+        make_shared<v1::Multiply>(norm, make_shared<v1::Subtract>(r, g)),
+        make_shared<v0::Constant>(element::f32, Shape{}, std::vector<float>{4.0f / 6.0f})
+    );
+
+    // select hue based on the maximum component
+    // check if `r` is the max, otherwise check if `g` is the max, if not use `b`'s hue
+    auto hh = make_shared<v1::Select>(
+        r_eq_v,
+        hue_case_r,  // Use hue_case_r if r is max
+        make_shared<v1::Select>(
+            g_eq_v,
+            hue_case_g,  // Use hue_case_g if g is max
+            hue_case_b   // Use hue_case_b otherwise (b is max)
+        )
+    );
+
+    // range = 0.0: hh = 0
+    auto hh_zero_range = make_shared<v1::Select>(make_shared<v1::Equal>(range, zero), zero, hh);
+
+    // hh < 0.0: hh = hh + 1
+    auto hh_final = make_shared<v1::Select>(make_shared<v1::Less>(hh, zero), make_shared<v1::Add>(hh_zero_range, one), hh_zero_range);
+
+    return make_shared<tuple<shared_ptr<Node>, shared_ptr<Node>, shared_ptr<Node>>>(hh_final, ss, vv);
+}
+
+shared_ptr<Node> hsv_to_rgb(shared_ptr<Node> h, shared_ptr<Node> s, shared_ptr<Node> v, element::Type type) {
+    // image format conversion based on https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/image/adjust_saturation_op.cc
+    // c = s * v;
+    auto c = make_shared<v1::Multiply>(s, v);
+    // m = v - c;
+    auto m = make_shared<v1::Subtract>(v, c);
+    // dh = h * 6;
+    auto dh = make_shared<v1::Multiply>(h, make_shared<v0::Constant>(type, Shape{}, 6.0f));
+
+    // fmodu rounded to within [0, 2)
+    auto fmodu = make_shared<v1::FloorMod>(dh, make_shared<v0::Constant>(type, Shape{}, 2.0f));
+
+    //  x = c * (1 - std::abs(fmodu - 1));
+    auto x = make_shared<v1::Multiply>(c, make_shared<v1::Subtract>(make_shared<v0::Constant>(element::f32, Shape{}, 1.0f), make_shared<v0::Abs>(make_shared<v1::Subtract>(fmodu, make_shared<v0::Constant>(type, Shape{}, 1.0f)))));
+
+    // h_category = static_cast<int>(dh);
+    auto h_category = make_shared<v0::Convert>(make_shared<v0::Floor>(dh), element::i32);
+
+    auto zeros = make_shared<v0::Constant>(type, x->get_shape(), 0.0f);
+
+    auto rr_options = NodeVector{c, x, zeros, zeros, x, c};
+    auto gg_options = NodeVector{x, c, c, x, zeros, zeros};
+    auto bb_options = NodeVector{zeros, zeros, x, c, c, x};
+
+    auto rr_concat = make_shared<v0::Concat>(rr_options, -1); 
+    auto gg_concat = make_shared<v0::Concat>(gg_options, -1);
+    auto bb_concat = make_shared<v0::Concat>(bb_options, -1);
+    
+    // use a gather operation to select the correct channel values based on h_category
+    int batch_dim = s->get_shape().size() - 1;
+
+    auto axis = make_shared<v0::Constant>(element::i32, Shape{}, -1);
+    auto rr = make_shared<v8::Gather>(rr_concat, h_category, axis, batch_dim);
+    auto gg = make_shared<v8::Gather>(gg_concat, h_category, axis, batch_dim);
+    auto bb = make_shared<v8::Gather>(bb_concat, h_category, axis, batch_dim);
```

Comment:
please re-write thic code by using `Select` operation for `switch-case implementation` because I am afraid that `Gather` with dynamic `axis` is not supported by OV that are needed for general case.
In `Select` please compute `concat_rr_gg_bb` by condition `h_category`. After that add `m` and it looks all. Other code below is not needed

Reply:
Axis in gather is a constant, only batch-dim is dynamic but it is an integer argument. 
 If using Select ops, it would require about 5 Select and related condition tensors, right?

Reply:
great catch. I was wrong, I looked sleepy when I left a comment:)
Let us proceed with you solution.